pkg/report/linux.go

```diff
@@ -661,3 +711,93 @@ var linuxOopses = []*oops{
 		[]*regexp.Regexp{},
 	},
 }
+
+func isCorrupted(desc string, text string) bool {
+	if !strings.Contains(text, "Call Trace") && !strings.Contains(text, "backtrace") {
+		// Text must contain 'Call Trace' or 'backtrace'.
+		return true
+	}
+	corruptedDescRegExps := []*regexp.Regexp{
+		// 'kernel panic: Fatal exception' is usually printed after BUG,
+		// so if we captured it as a report description, that means the
+		// report got truncated and we missed the actual BUG header.
+		regexp.MustCompile(`kernel panic: Fatal exception`),
+		// Same, but for WARNINGs and KASAN reports.
+		regexp.MustCompile(`kernel panic: panic_on_warn set`),
+		// Sometimes timestamps get merged into the middle of report description.
+		regexp.MustCompile(`\[ *[0-9]+\.[0-9]+\]`),
+		regexp.MustCompile(`\[ *[0-9]+\.NUM\]`),
+		regexp.MustCompile(`\[ *NUM\.NUM\]`),
+		regexp.MustCompile(`\[ *NUM\.[0-9]+\]`),
+	}
+	for _, re := range corruptedDescRegExps {
+		if re.MatchString(desc) {
+			return true
+		}
+	}
+	corruptedDescStrings := []string{
+		// Sometimes with such BUG failures, the second part of the header doesn't get printed
+		// or gets corrupted, because kernel prints it as two separate printk() calls.
+		"BUG: unable to handle kernel",
+		// If we failed to extract function name where the fault happened,
+		// the report is most likely truncated.
+		"general protection fault",
+		"BUG: bad unlock balance",
+		"divide error",
+		"invalid opcode",
+	}
+	for _, s := range corruptedDescStrings {
+		if strings.TrimSpace(desc) == s {
+			return true
+		}
+	}
+	corruptedTextRegExps := []*regexp.Regexp{
+		// If report text contains 'printk messages dropped' it is most likely corrupted.
+		regexp.MustCompile(`printk messages dropped`),
+	}
+	for _, re := range corruptedTextRegExps {
+		if re.MatchString(text) {
+			return true
+		}
+	}
+	crashTypes := []string{"BUG", "WARNING", "INFO", "KASAN", "KMSAN", "UBSAN"}
```

Comment:
Tests pass without this part and I am struggling to understand when this can fire.
Either add test or remote this.

Reply:
Removed in #427.